lambda/upload/handler/dyQueries.go

```diff
@@ -87,14 +89,18 @@ func (q *UploadDyQueries) GetUploadFiles(entries []UploadEntry) ([]uploadFile.Up
 		dbItems := dbResults.Responses[ManifestFileTableName]
 
 		// Re-request missing items if for some reason dynamodb does not return all events
+		retryCount := 1
 		for len(dbResults.UnprocessedKeys) > 0 {
-			moreResults, err := q.db.BatchGetItem(context.Background(), &batchItemInput)
+			exponentialWaitWithJitter(retryCount)
+			dbResults, err = q.db.BatchGetItem(context.Background(), &dynamodb.BatchGetItemInput{
+				RequestItems: dbResults.UnprocessedKeys,
+			})
 			if err != nil {
 				log.Error(fmt.Sprintf("Unable to get dbItems: %v", err))
 				return nil, nil, err
 			}
-
-			dbItems = append(dbItems, moreResults.Responses[ManifestFileTableName]...)
+			dbItems = append(dbItems, dbResults.Responses[ManifestFileTableName]...)
+			retryCount++
 		}
```

Comment:
is it worth adding a `maxRetries` to the loop conditional too in case batch get item always returns the same unprocessed items for some weird reason?

Reply:
Yes, thanks! I've added a `maxRetries` now. Instead of returning an error, it just logs a warning that there were unprocessed keys remaining. Does that seem okay?